chapter2/tests/t2/chapter2-t2-gen-kustomization.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+#
+# Task 2 - Generate kustomization.yaml
+set -eo pipefail
+
+
+repo_dir=repos
+
+
+# Function for printing error messages and exiting with a specific exit code
+# $1 - The exit code to use when exiting the script
+# $2 - The error message to print
+err() {
+  local code=$1
+  shift
+  echo "[[ERROR]: $(date +'%Y-%m-%dT%H:%M:%S%z')]: $*" >&2
+  exit "$code"
+}
+
+
+# Checking permissions
+if [[ ! -r "$(pwd)" ]]; then
+  err 255 'Directory is not readable.'
+fi
+
+if [[ "$#" -lt 1 ]]; then
+  err 255 'At least 1 or more args are expected'
+fi
+
+if [[ ! -d "$repo_dir" ]]; then
+  mkdir $repo_dir
+fi
+
+
+# ssh-keys generation
+for repo_name in "$@"; do
+  rm -f ./$repo_dir/$repo_name-deploy-key.pem* && ssh-keygen -q -t ed25519 -N "" -C "$repo_name" -f ./$repo_dir/$repo_name-deploy-key.pem;
+done
+
+
+# Generate file beginning
+echo "
```

Comment:
Use [cat << EOF](https://distroid.net/use-cat-eof-linux/) rather than echo, it's a conventional way of writing multi-line strings.

Reply:
> Use [cat << EOF](https://distroid.net/use-cat-eof-linux/) rather than echo, it's a conventional way of writing multi-line strings.

Ok, fixed